packages/react/src/components/RuleBuilder/Rule.jsx

```diff
@@ -0,0 +1,234 @@
+import * as React from 'react';
+import { Button, Dropdown, TextInput } from 'carbon-components-react';
+import { Add32, Subtract32, TextNewLine32 } from '@carbon/icons-react';
+import PropTypes from 'prop-types';
+
+import { settings } from '../../constants/Settings';
+import { TableColumnsPropTypes } from '../Table/TablePropTypes';
+
+import GroupLogic from './GroupLogic';
+
+const { iotPrefix } = settings;
+
+const defaultProps = {
+  i18n: {
+    all: 'ALL',
+    any: 'ANY',
+    lessThan: 'Less than',
+    lessThanOrEqual: 'Less than or equal to',
+    equals: 'Equals',
+    greaterThanOrEqual: 'Greater than or equal to',
+    greaterThan: 'Greater than',
+  },
+};
+
+export const GroupLogicPropType = PropTypes.oneOf(['ALL', 'ANY']);
+export const LogicPropType = PropTypes.oneOf(['LT', 'LTOET', 'EQ', 'GTOET', 'GT']);
+export const RulesPropType = PropTypes.shape({
+  id: PropTypes.string.isRequired,
+  column: PropTypes.string.isRequired,
+  value: PropTypes.string.isRequired,
+  logic: LogicPropType.isRequired,
+});
+
+export const RuleGroupPropType = PropTypes.shape({
+  id: PropTypes.string.isRequired,
+  groupLogic: GroupLogicPropType.isRequired,
+});
+
+RuleGroupPropType.rules = PropTypes.arrayOf(PropTypes.oneOf([RulesPropType, RuleGroupPropType]));
+
+const propTypes = {
+  rule: PropTypes.oneOfType([RulesPropType, RuleGroupPropType]).isRequired,
+  columns: TableColumnsPropTypes.isRequired,
```

Comment:
Why are we using the table column props here?  We should define our own properties for this.  Does the rule builder support the size property?  what properties do we expect to use here?

Reply:
@stuckless I don't know what you mean by "size property". I don't see that used in reference to the table anywhere. Can you point me in the right direction?

Reply:
My point is not that you should be using it but rather... are you using all the props from TableColumnPropTypes here, or just a couple.  ie, we should define just that props you are using.  I said "size" but I was thinking of "width" of just a random property on the column that you are likely not using (nor should you) but you define it as something that you accept.